src/main/java/com/ays/user/service/impl/UserSelfServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.ays.user.service.impl;
+
+import com.ays.auth.model.AysIdentity;
+import com.ays.user.model.dto.request.UserSupportStatusUpdateRequest;
+import com.ays.user.model.entity.UserEntity;
+import com.ays.user.repository.UserRepository;
+import com.ays.user.service.UserSelfService;
+import com.ays.user.util.exception.AysUserNotExistByIdException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+/**
+ * UserSelfServiceImpl is an implementation of the UserSelfService interface.
+ * It manages a user's own operations.
+ */
+@Service
+@RequiredArgsConstructor
+public class UserSelfServiceImpl implements UserSelfService {
```

Comment:
Bu sınıfı dışarıdan çağırmayacağımız için package-private yapabiliriz

Reply:
Gerekli düzenlemeyi yaptım.